src/widgets.rs

```diff
@@ -444,7 +444,13 @@ impl Widget for &str {
 /// [`Rect`].
 impl WidgetRef for &str {
     fn render_ref(&self, area: Rect, buf: &mut Buffer) {
-        buf.set_string(area.x, area.y, self, crate::style::Style::default());
+        buf.set_stringn(
+            area.x,
+            area.y,
+            self,
+            area.width as usize,
+            crate::style::Style::default(),
```

Comment:
I am curious why it's not imported in the file with a use and rather used with a full path here.

(Also I would go with the `const fn` `Style::new()` but thats not important for this PR.)

Reply:
I just left that as I found it. Can change it if you want.

Reply:
I am mainly curious why it was that way to begin with… What the intention of that was.

Reply:
Don't think I can help you with that one. 

Changed it, looks nicer now as it's one row again.

Reply:
(I mainly asked that question towards other @ratatui-org/maintainers)

Reply:
Likely just an autocomplete that wasn't noticed.